emp/custom-registry/emp_image_helper.sh

```diff
@@ -0,0 +1,350 @@
+CR="$(tput sgr0 2>/dev/null || /bin/true)"                                                # RESET
+
+# Spinner animation
+spinner() {
+    local pid=$1
+    local delay=0.1
+    local spinstr='|/-\'
+    local temp=""
+    local image=""
+
+    if [ -f "$IMG_FILE" ]; then
+        debug "File $IMG_FILE exists."
+        image=`cat $IMG_FILE`
+    else
+        image=$IMG_FILE
+    fi
+    last_word_length=0
+    max=0
+    while kill -0 "$pid" 2>/dev/null; do
+        temp="${spinstr#?}"
+        spinstr=$temp${spinstr%"$temp"}
+        image=`cat $IMG_FILE`
+        printf "\r${Cb} ⋯ %s  ${Cy}Processing: %*s %s ${CR}" "${spinstr:0:1}" "$last_word_length" "$image" >&2
+        last_word_length=${#IMG_FILE}
+        if (( last_word_length > max )); then
+            max=$last_word_length
+        fi
+        sleep "$delay"
+    done
+    max=$(expr $max + 32)
+    printf "\r${Cgr}%s ✓ Completed %*s ${CR}\n" "$(date +'%F %T')" "$max" >&2
+}
+
+# debug - prints a DEBUG message (normally gray on black)
+debug() {
+    [ $VERBOSE -le 0 ] || echo "$(date +'%F %T') $Cg DEBUG: $* $CR" >&2
+}
+
+# info - prints an INFO message (normally white on blue)
+info() {
+    echo "$(date +'%F %T') $Cb INFO: $* $CR" >&2
+}
+
+# warn - prints a WARN message (normally yellow on black)
+warn() {
+    echo "$(date +'%F %T') $Cy WARN: $* $CR" >&2
+}
+
+# fail - prints a FATAL message (yellow on red) and exits the script
+fail() {
+    local args=${*:-error}
+    echo "$(date +'%F %T') $Cr FATAL $args $CR" >&2
+    trap - EXIT
+    exit 2
+}
+
+# SUPPORTING FUNCTIONS
+
+exit_trap() {
+    local rc=$?
+    if [ $rc -eq 0 ]; then
+        debug "END"
+    else
+        warn "$0 FAILED (rc=$rc)"
+    fi
+    exit $rc
+}
+trap exit_trap EXIT
+
+# dummy_cmd - replacement for docker/kubelet to display parameters/output
+dummy_cmd() {
+    if echo "$*" | grep -q -- '-$'; then
+        echo "DRY-RUN: $* << EOF" >&2
+        cat >&2
+        echo EOF >&2
+    else
+        echo "DRY-RUN: $*" 2>&1
+    fi
+}
+
+# num_images returns a number of images
+num_images() {
+    # shellcheck disable=SC2086
+    echo $IMAGES | wc -w
+}
+
+trim_space() {
+    local str=$1
+    # Trim leading and trailing spaces using parameter expansion
+    trimmed_str="${str#"${str%%[![:space:]]*}"}"  # Trim leading spaces
+    trimmed_str="${trimmed_str%"${trimmed_str##*[![:space:]]}"}"  # Trim trailing spaces
+    echo "$trimmed_str"
+}
+
+# pull_images - pulls container images with spinner
+pull_images() {
+    local num_img    
+    num_imgs=$(num_images)
+    num_img=$(trim_space $num_imgs)
+    info "Pulling $num_img images for Elastic Machine Pool by Platform9"
+    
+    # Temporary file for output
+    local tmpfile
+    tmpfile=$(mktemp)
+    debug "Logs are in $tmpfile"
+    # Async pull with spinner
+    {
+        for img in $IMAGES; do
+            echo "$img" > $IMG_FILE
+            info "Getting $img"
+            $CNT_CMD pull "$img" 2>&1
```

Comment:
we should add `--platform linux/amd64` in this command. This is due to the fact that while pulling docker images, the docker pulls the images according to the underlying architecture and as a result the pods will crash if they are of different arch., and not of linux/amd64.